spec/models/concerns/energy_tariff_holder_spec.rb

```diff
@@ -40,13 +40,15 @@
       it 'maps the tariffs to meter attribute' do
         expect(tariff_holder.energy_tariff_meter_attributes.first).to be_a MeterAttribute
       end
-
     end
 
     context 'with School' do
-      let!(:site_settings)   { SiteSettings.current }
-      let(:school_group)     { nil }
-      let(:tariff_holder)    { create(:school, school_group: school_group) }
+      let!(:site_settings) { SiteSettings.current }
+      let(:school_group) { nil }
+      let(:tariff_holder) { create(:school, school_group: school_group) }
+      let!(:energy_tariff_electricty_half_hourly) { create(:energy_tariff, :with_flat_price, tariff_holder: tariff_holder, meter_type: "electricity", applies_to: "half_hourly", name: 'Electricity Tariff half_hourly') }
```

Comment:
Because these three tariffs are only used in one of the specs, its clear if you create a separate context.

E.g.

```
context 'when filtering tariffs' do
 let!(:energy_tariff_electricty_half_hourly) { create(:energy_tariff, :with_flat_price, tariff_holder: tariff_holder, meter_type: "electricity", applies_to: "half_hourly", name: 'Electricity Tariff half_hourly') }
 ...
end
```